### PR TITLE
Adds Doomguy and energy chainsaws

### PR DESCRIPTION
**Pull request by genessee596**
**THE WHAT**
This PR adds a new ERT - 'The Juggernaut' - which acts like a one-man deathsquad, armed to the teeth (read: pulse destroyer and energy chainsaw) and (presumably) extremely robust. Any resemblance to other person or persons, real or fiction, is purely coincidental. Totally.

Also, adds a new weapon that can only be found in the hands of the Juggernaut -- an Energy Chainsaw, which deals 32 brute damage and pierces all armor. Also, knocks victims down for half a second per hit. Also, it's quite small compared to a normal chainsaw, and can be stored in bags.

Also also, adds a special set of deathsquad armor called 'Juggernaut Armor', which has a hefty shield and some of the best armor in the game. And it totally looks like classic doomguy. Which is awesome.

EDIT: Also, forgot to mention that he gets Krav Maga in the rare case he somehow loses his chainsaw.

Sprites;
image

**THE HOW**
Adding a new ERT in the various ERT.dm files, plus sprites in the various .dmi files edited, made this a relatively easy task. 

The chainsaw was a pain, but I managed to get it working with the help of the fine folks on discord. Thanks, everyone who helped answer my questions!

**THE WHY**
It's cool as hell to go complete apeshit on a crowd of revs as a deathsquaddie (and it's pretty fun to watch too), but it's even cooler to see a solo man wipe out a horde of lings or revs. 

Plus, a little bit of variation in ERTs is cool too.

The energy chainsaw is a neat thing -- more chainsaws, the better, right? -- and I could see it falling in the hands of a syndicate nuclear uplink in the future.

:cl:
add: The Juggernaut, a one-man ERT with the sole purpose to wipe out every living thing on the station. Coincidentally looks like Doomguy.
add: Energy Chainsaws, found only in the hands of the Juggernaut. Very strong weapon, and can be used to cut loaves of bread to boot!
/:cl:

### CODE REVIEW

**Comment by Krysonism** [comment]
Cool PR but I dont like how the shield obscures so much of the hardsuit sprite.

**Comment by PepperPrepper** [comment]
No classic armor? no ssg? THIS PR IS UNFAITHFUL!!!!1111!

very good pr 10/10 one of the only times you will never see me say speedmerge

**Comment by genessee596** [comment]
Commit changes:

Made the Doomguy Hardsuit less OP ( shield from 6 max with a recharge every 1.5 secs to 1 max with a recharge of 10 secs)

Made the Energy Chainsaw less OP ( item size Small -> Bulky, damage decreased from 50 -> 32)

**Comment by AnonymousNow** [comment]
If you're going to make an energy chainsaw, you need an orange, transparent variant.

File: code/game/objects/items/twohanded.dm [hunk 1]
```diff
@@ -613,6 +613,28 @@
...
 		. = ..()
 
+// ENERGY CHAINSAW!!
+/obj/item/twohanded/required/chainsaw/energy
+	name = "energy chainsaw"
+	desc = "As awesome as both an energy sword and chainsaw combined."
+	icon_state = "energy_chainsaw_off"
+	force_on = 32
+	armour_penetration = 100
+	var/knockdown = 1
```
Comment (Couls) [inline code/game/objects/items/twohanded.dm]: unused var, you can just get rid of this

**Comment by wesoda25** [comment]
`Verified`

**Comment by CRITAWAKETS** [comment]
You forgot to push the item sprite for the chainsaw, along with the item sprites for the doomguy suit and head respectively.

**Comment by HomocidlePancakes** [comment]
Power armor is for pussies.

**Comment by Ziiro** [comment]
Suggestions, if you want the real Doomslayer experience:

* Drop the Shield
* Immune to Stun
* Immune to Slowdown

This will actually make it much more threatening than normal Deathsquad, especially with a melee weapon.

I would also suggest for academic reasons the kit includes a pre sawed-off double barreled shotgun.

**Comment by GuyonBroadway** [comment]
Oh hey we had this shit on hippie for years now.

https://github.com/HippieStation/HippieStation/pull/3101

**Comment by Couls** [comment]
weird flex but ok

**Comment by TheVekter** [comment]
**WHO'S A MAN AND A HALF
I'M A MAN AND A HALF

BERZERKER PACKIN' MAN AND A HALF
THERE'S NOTHING WRONG WITH YOU I CAN'T FIX
WITH MY FISTS!**

**Comment by Dorsisdwarf** [comment]
I feel like the doomguy is gonna get absolutely bodied if hes basically just a dual-esworder but only blocks *one* shot before going down.

**Comment by ATH1909** [comment]
Yeah, make him stun/knockdown/slip immune, but get rid of that really obscuring shield.

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -856,3 +856,30 @@
...
+/obj/item/clothing/suit/space/hardsuit/shielded/doomguy
+	name = "juggernaut armor"
+	desc = "A somehow spaceworthy set of armor with outstanding protection against almost everything. Comes in an oddly nostalgic green. "
+	icon_state = "doomguy"
+	item_state = "doomguy"
+	hardsuit_type = "syndi"
+	max_charges = 1
+	current_charges = 1
+	recharge_delay = 100
+	armor = list("melee" = 90, "bullet" = 90, "laser" = 70, "energy" = 70, "bomb" = 100, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)
```
Comment (ATH1909) [inline code/modules/clothing/spacesuits/hardsuit.dm]: You might as well crank these values up to something absurd like 200, to make armor-piercing weapons less effective.

Don't worry, effective armor values (after armor_penetration is applied) cap out at 90, so you're not gonna make Doomguy immune to almost every attack in the game or something.

File: code/modules/clothing/outfits/ert.dm [hunk 3]
```diff
@@ -333,3 +333,29 @@
...
 	r_hand = /obj/item/megaphone
 	head = /obj/item/clothing/head/intern
+
+/datum/outfit/doomguy
+	name = "The Juggernaut"
+
+	uniform = /obj/item/clothing/under/rank/centcom/officer
+	suit = /obj/item/clothing/suit/space/hardsuit/shielded/doomguy
+	shoes = /obj/item/clothing/shoes/jackboots/fast
+	gloves = /obj/item/clothing/gloves/krav_maga/combatglovesplus
```
Comment (ATH1909) [inline code/modules/clothing/outfits/ert.dm]: might wanna mention that Doomguy gets krav maga in the PR description

**Comment by KomradeSpectre** [comment]
@genessee596 I'm not against having doomguy, but those energy chainsaws look like vibrators.

**Comment by genessee596** [comment]
> You forgot to push the item sprite for the chainsaw, along with the item sprites for the doomguy suit and head respectively.

image
No, I didn't.

**Comment by genessee596** [comment]
> @genessee596 I'm not against having doomguy, but those energy chainsaws look like vibrators.

even more deadly...

**Comment by genessee596** [comment]
> Suggestions, if you want the real Doomslayer experience:
> 
> * Drop the Shield
> * Immune to Stun
> * Immune to Slowdown
> 
> This will actually make it much more threatening than normal Deathsquad, especially with a melee weapon.
> 
> I would also suggest for academic reasons the kit includes a pre sawed-off double barreled shotgun.

Good news, his jackboots are the same ones that Highlanders get, making him immune to slowdown! I'm totally okay with making him immune to stuns and dropping the shield, but I don't know how -- if you want to commit that for me, I'd happily verify it.

Sure, I'll replace the pulse destroyer with a double barreled shotgun.

File: code/game/objects/items/twohanded.dm (hunk 1 above)
Comment (genessee596) [reply code/game/objects/items/twohanded.dm]: I thought 'var/knockdown = 1' lets the chainsaw actually knock people down?

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 2 above)
Comment (genessee596) [reply code/modules/clothing/spacesuits/hardsuit.dm]: I will do that!

File: code/modules/clothing/outfits/ert.dm (hunk 3 above)
Comment (genessee596) [reply code/modules/clothing/outfits/ert.dm]: Done

**Comment by genessee596** [comment]
Commit changes:

Bumped Doomguy armor values WAAAAAY up (read: 200 for melee, bullet, energy and laser) because someone said so

**Comment by genessee596** [comment]
Commit changes:

Doomguy now starts with a Double Barrelled Shotgun on his back instead of a pulse destroyer. 

Shotgun can be sawn off with the Energy Chainsaw, I think?
He can get spare shells from the ERT armory

**Comment by HomocidlePancakes** [comment]
Can we get some sort of a botomless ammobag for the shottie? We already have infinite lockers.

**Comment by carlarctg** [comment]
200 melee 😂

**Comment by TheVekter** [comment]
> 200 melee

He has a fucking Berserker pack! WHO'S A MAN AND A HALF?! RIP AND TEAR! RIP AND TEAR YOUR GUTS!

**Comment by carlarctg** [comment]
i'd like it if it was actually fun to fight the doomguy instead of punching a tank

if you give him retarded numbers like 200 melee just give him godmode lol

**Comment by genessee596** [comment]
> i'd like it if it was actually fun to fight the doomguy instead of punching a tank
> 
> if you give him retarded numbers like 200 melee just give him godmode lol

Doomguy is only meant to exist when the entire station is completely overrun with enemies -- to the point that only a well-equipped deathsquad could have any hope of winning in a fight.

Or when the admins are bored. I ain't the cops.

Basically, unless Doomguy is staggeringly unrobust, you're doomed to lose if he sees you. Sorry.

**Comment by genessee596** [comment]
> Can we get some sort of a botomless ammobag for the shottie? We already have infinite lockers.

I'd do that if I had any idea how. Anyways, you'll have plenty off ammunition (or alternate weapons) from the ERT armory, so ammo isn't really a problem. Doomguy also has ERT commander access, so he'll be able to raid station armory if he needs more ammo.

**Comment by TheVekter** [comment]
Okay so memes aside, realistically, you want the Doom Marine to be able to beat Bubblegum without too much trouble. I think from there you balance it a bit further out. It should also probably be made plain that he isn't the kind of thing you trade TCs for and spawning him into the round will *completely* fuck it up.

**Comment by cacogen** [comment]
these sprites suck (not that I can sprite)

**Comment by genessee596** [comment]
@TheVekter

Doomguy gets 200 melee defense on his armor (plus a recharging shield) making him basically invulnerable to almost everything. A complete immunity to slowdown via his boots, and a 32 force melee weapon with garuanteed knockdown on hit plus a decent shotgun means that only the least robust of doomguys couldn't kill bubblegum.

**Comment by genessee596** [comment]
@cacogen

yes your feedback is very helpful please continue spreading your infinite knowledge to humble prs such as mine

**Comment by kriskog** [comment]
Merge conflicts in the dmi's. Resolve, but make sure you dont overwrite the changes to master.

**Comment by Couls** [comment]
make sure you don't lose your changes as well

**Comment by cacogen** [comment]
Even just steal the Hippie sprites

**Comment by optimumtact** [comment]
If your feedback boils down to "this sucks" just leave an emojii downvote and move on.

If you have actually useful constructive criticism then comment with that first.

**Comment by cacogen** [comment]
The chainsaw sprites aren't very good and there should be more quality control on the sprites that get into the game. As GuyonBroadway said, Hippie already has a version of this feature, and after looking at their chainsaw sprites for it, they're  better. Also the item sprites (not the in-hands) for this seem to be missing.

Hippie
Hippie echainsaw item sprite
Hippie echainsaw right in-hand

this PR
tg echainsaw right in-hand

**Pull request closed without merging**